modules/files/serviceLevel_endpoint.bal

```diff
@@ -0,0 +1,243 @@
+////Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/jsonutils as jsonlib;
+import ballerina/lang.'string as stringLib;
+
+public client class ServiceLevelClient {
+    private string sharedKeyOrSASToken;
+    private string baseUrl;
+    private http:Client httpClient;
+    private boolean isSharedKeyUsed = false;
+    private AzureConfiguration azureConfig;
+
+    # Initalize Azure Client using the provided azureConfiguration by user
+    #
+    # + azureConfig - AzureConfiguration record
+    public function init(AzureConfiguration azureConfig) returns error? {
+        http:ClientSecureSocket? secureSocketConfig = azureConfig?.secureSocketConfig;
+        self.sharedKeyOrSASToken = stringLib:substring(azureConfig.sharedKeyOrSASToken, startIndex = 1);
```

Comment:
Can be simplified to

self.sharedKeyOrSASToken = azureConfig.sharedKeyOrSASToken.substring(startIndex, endIndex);